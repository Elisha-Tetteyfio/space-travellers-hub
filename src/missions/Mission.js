import React from 'react';

/* eslint-disable react/prop-types */
function Mission({ name, id, description }) {
  return (
    <li id={id}>
      <div>{name}</div>
      <div>{description}</div>
      <div><button type="button">Status</button></div>
      <div><button type="button">Join Mission</button></div>
    </li>
  );
}

export default Mission;
