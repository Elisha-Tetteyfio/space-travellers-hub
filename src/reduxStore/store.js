import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
  },
});

export default store;
