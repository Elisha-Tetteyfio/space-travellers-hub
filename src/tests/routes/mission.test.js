import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Mission from '../../missions/Mission';
import store from '../../reduxStore/store';

describe('Testing mission component', () => {
  it('Test mission', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Mission />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
