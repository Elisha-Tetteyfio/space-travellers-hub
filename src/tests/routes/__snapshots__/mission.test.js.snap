// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing mission component Test mission 1`] = `
<tr>
  <th />
  <td />
  <td>
    <p
      className="notMember"
    >
      NOT A MEMBER
    </p>
  </td>
  <td>
    <div>
      <button
        className="joinBtn"
        onClick={[Function]}
        type="button"
      >
        Join Mission
      </button>
    </div>
  </td>
</tr>
`;
